### ISSUE TITLE
Unable to json_encode or iterate over variable after setting value equal to _context

### ISSUE THREAD

**Issue by scholtron** [twigjs/twig.js#341]
I am attempting to create a template which can generically take in input and render a generic table.

Here is an example dataset and render:
```javascript
var data = [1,2,3,4,5,69];
var data2 = {derp: data};
var output = twig({ ref: 'exampleTemplate' }).render(data2);
```

Since I do not know what the initial input might be, I am attempting to set a variable initially to _context:
```
{% set list = _context %}
```

Attempting to iterate over the content of the "list" variable using `{{ list|json_encode }}` or `{% for rows in list %}{{ rows|json_encode }}{% endfor %}` results in one of two errors:

RangeError: Maximum call stack size exceeded
Converting circular structure to JSON

Looking further into the contents of "list", it has the keys of "derp" and "list":
```list|keys|json_encode = ["derp","list"]```

It seems that the list variable is passed by referenced in someway which is causing the circular error.

If I use the same template in Twig PHP it does not pass by reference and instead, there is only one key "derp".

I had seen https://github.com/twigjs/twig.js/pull/215 and I couldnt tell if this was by design or not.

Am I doing the initial set correctly?

I could change my initial render code (see below) but am interested in how _context should be used.
```javascript
var output = twig({ ref: 'exampleTemplate' }).render({ list: data2 });
```

Thank you!

**Comment by evgenius** [comment]
@scholtron Thanks for catching it!

**Comment by dave-irvine** [comment]
@scholtron @evgenius I think I've fixed this in #345 but I'd appreciate some comments on it.

**Comment by scholtron** [comment]
Looks good to me, however I'm a bit green to the Twig internals.
